config/configtls/clientcasfilereloader.go

```diff
@@ -0,0 +1,148 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configtls // import "go.opentelemetry.io/collector/config/configtls"
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+type clientCAsFileReloader struct {
+	ClientCAsFile   string
+	certPool        *x509.CertPool
+	lastReloadError error
+	lock            sync.RWMutex
+	loader          clientCAsFileLoader
+	watcher         *fsnotify.Watcher
+	shutdownCH      chan bool
+}
+
+type clientCAsFileLoader interface {
+	loadClientCAFile() (*x509.CertPool, error)
+}
+
+func newClientCAsReloader(clientCAsFile string, loader clientCAsFileLoader) (*clientCAsFileReloader, error) {
+	certPool, err := loader.loadClientCAFile()
+	if err != nil {
+		return nil, fmt.Errorf("failed to load TLS config: failed to load client CA CertPool: %w", err)
```

Comment:
This looks like two errors in one :-) How about just the second part, which is more specific? 

```suggestion
		return nil, fmt.Errorf("failed to load client CA CertPool: %w", err)
```

Reply:
This seems to be an expected pattern, these tests are failing otherwise [configgrpc_test.go#L242](https://github.com/open-telemetry/opentelemetry-collector/blob/12c782d9c7883759246bd516bb202e753bd5b202/config/configgrpc/configgrpc_test.go#L242)

Reply:
You can keep it like that, but you could also just change the test, no?